btrace-agent/src/main/java/org/openjdk/btrace/agent/Main.java

```diff
@@ -184,7 +185,7 @@ private static synchronized void main(String args, Instrumentation inst) {
 
   private static boolean hasScripts() {
     return argMap.containsKey(SCRIPT)
-        || argMap.containsKey(SCRIPT_OUTPUT_DIR)
+        || argMap.containsKey(SCRIPT_DIR)
         || argMap.containsKey(SCRIPT_OUTPUT_FILE);
```

Comment:
This seems like only `SCRIPT` and `SCRIPT_DIR` should be checked here.
The `hasScripts()` is used to enable the `noServer` option by default so it seems right when it would not be affected by the output dir argument.